source/server/converters.h

```diff
@@ -341,10 +342,10 @@ inline void convert_to_grpc(const SmtSpectrumInfoType& input, nidevice_grpc::Smt
 }
 
 const int64 SecondsFromCVI1904EpochTo1970Epoch = 2082844800LL;
+const int64 SecondsFromDAQmx0001EpochToCVI1904Epoch = -((static_cast<int64>(0xfffffff2) << 32) | 0x0493b980); // extracted from NITYPES_ABSOLUTETIME_EPOCH_BIAS_FROM_0001 in ni-central/src/platform_services/abstractions/niatomicd/nitypes/source/nitypes/time/AbsoluteTime.h
```

Comment:
0xFFFFFFF2_0493b980 is negative, so you negate to get the positive number of seconds from Y0001 to Y1904.

Then when you use it:

> const int64 dot_net_ticks_since_1904 = dot_net_ticks - SecondsFromDAQmx0001EpochToCVI1904Epoch * DotNetTicksPerSecond;

dot_net_ticks is the ticks since Y0001, which includes the ticks between Y0001 and Y1904, so you subtract the number of ticks between Y0001 and Y1904. Ok, I think this makes sense.